Content.Server/Administration/PlaytimeManager.cs

```diff
@@ -0,0 +1,180 @@
+
+    bool Ticking { get; set; }
+
+    public void TryLoadPlaytime(IPlayerSession session, PlayerData data);
+}
+
+public class PlaytimeManager : IPlaytimeManager
+{
+    [Dependency] private readonly IPlayerManager _playerManager = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IServerDbManager _dbManager = default!;
+    [Dependency] private readonly IAfkManager _afkManager = default!;
+
+    /// <summary>
+    ///     Tracks time between updates. See <see cref="UpdateInterval"/>
+    /// </summary>
+    public float SecondsSinceUpdate { get; set; } = 0f;
+
+    /// <summary>
+    ///     Number of seconds between each <see cref="UpdatePlaytimes"/>. Defaults to 10 minutes.
+    /// </summary>
+    public float UpdateInterval { get; set; } = 600f;
+
+    /// <summary>
+    ///     Whether playtime tracking is active
+    /// </summary>
+    public bool Ticking { get; set; } = true;
+
+    public void Update(float deltaTime)
+    {
+        if (!Ticking) return;
+
+        SecondsSinceUpdate += deltaTime;
+
+        if (SecondsSinceUpdate >= UpdateInterval)
+        {
+            UpdatePlaytimes();
+            SecondsSinceUpdate = 0;
+        }
+    }
+
+    /// <summary>
+    ///     Called during round end to ensure playtime isn't lost.
+    /// </summary>
+    public void Shutdown()
+    {
+        var initialUpdateInterval = UpdateInterval;
+
+        // Give it one last update and then pause until the next round begins.
+        Ticking = false;
+        //This Max() should never be necessary but better safe than sorry.
+        UpdateInterval = Math.Max(UpdateInterval -= SecondsSinceUpdate, 0);
+        UpdatePlaytimes();
+
+        // Reset everything for next round.
+        UpdateInterval = initialUpdateInterval;
+        SecondsSinceUpdate = 0;
+    }
+
+    /// <summary>
+    ///     Returns the living and total playtime for a <see cref="IPlayerSession"/> in minutes.
+    /// </summary>
+    public void FetchPlaytime(IPlayerSession session, out int livingTime, out int totalTime)
+    {
+        var data = session.ContentData();
+        if (data is null)
+        {
+            livingTime = 0;
+            totalTime = 0;
+            return;
+        }
+        livingTime = data.LivingPlaytime;
+        totalTime = data.TotalPlaytime;
+    }
+
+    /// <summary>
+    ///     Handles actual playtime updates in the <see cref="PlayerData"/> and database.
+    /// </summary>
+    private void UpdatePlaytimes()
+    {
+        var intervalSpan = TimeSpan.FromSeconds(UpdateInterval);
+
+        foreach (IPlayerSession session in _playerManager.GetAllPlayers())
+        {
+            var data = session.ContentData();
+            if (data is null) continue;
+
+            switch (session.Status)
+            {
+                // Don't let them lose time between last update and disconnect
+                case SessionStatus.Disconnected:
```

Comment:
If you intended `GetAllPlayerData()` instead this would still be a bad idea because you really shouldn't be polling for every person that has ever connected to the server since it last restarted.